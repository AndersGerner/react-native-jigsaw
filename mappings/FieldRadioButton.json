{"name":"Field Radio Button","tag":"FieldRadioButton","category":"field","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Field_Radio.png","props":{"title":{"label":"Title","description":"Title to display alongside radio button","required":false,"editable":true,"value":"Title","type":"string"},"selected":{"label":"Selected","description":"Whether the radio button is selected","required":true,"editable":true,"value":true,"type":"boolean"},"color":{"label":"Color","description":"Color for the button","required":false,"editable":true,"value":"primary","type":"color"},"disabled":{"label":"Disabled","description":"Whether radio button is disabled","editable":true,"required":false,"value":null,"type":"boolean"}},"layout":{}}