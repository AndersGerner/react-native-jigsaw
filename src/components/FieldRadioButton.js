/* @flow */

import * as React from "react";
import { View, Text } from "react-native";
import color from "color";
import { withTheme } from "../core/theming";
import Touchable from "./Touchable";
import RadioButton from "./RadioButton";
import type { Theme } from "../types";
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types";

type Props = {|
  /**
   * Title to display alongside radio button
   */
  title?: string,
  /**
   * Status of radio button
   */
  selected: boolean,
  /**
   * Whether the radio button is disabled.
   */
  disabled?: boolean,
  /**
   * Function to execute on press.
   */
  onPress?: () => mixed,
  /**
   * @optional
   */
  theme: Theme
|};

class FieldRadioButton extends React.Component<Props> {
  onPress = () => {
    const { onPress } = this.props;

    if (onPress) {
      onPress();
    }
  };

  render() {
    const {
      title,
      selected,
      disabled,
      onPress,
      theme: { colors, typography, spacing, disabledOpacity }
    } = this.props;

    let titleColor = selected ? colors.medium : colors.light;

    if (disabled) {
      titleColor = color(titleColor)
        .alpha(disabledOpacity)
        .rgb()
        .string();
    }

    return (
      <Touchable onPress={this.onPress} disabled={disabled}>
        <View style={{ flexDirection: "row" }}>
          <RadioButton selected={selected} disabled={disabled} />
          <Text
            style={[
              typography.body1,
              { color: titleColor, marginLeft: spacing.medium }
            ]}
          >
            {title}
          </Text>
        </View>
      </Touchable>
    );
  }
}

export default withTheme(FieldRadioButton);

export const SEED_DATA = {
  name: "Field Radio Button",
  tag: "FieldRadioButton",
  category: COMPONENT_TYPES.button,
  preview_image_url:
    "https://res.cloudinary.com/altos/image/upload/v1541096663/draftbit/library/jigsaw-1.0/reps/Field_Radio.png",
  props: {
    title: {
      label: "Title",
      description: "Title to display alongside radio button",
      required: false,
      editable: true,
      value: "Title",
      type: FORM_TYPES.string
    },
    selected: {
      label: "Selected",
      description: "Whether the radio button is selected",
      required: true,
      editable: true,
      value: true,
      type: FORM_TYPES.boolean
    },
    disabled: {
      label: "Disabled",
      description: "Whether radio button is disabled",
      editable: true,
      required: false,
      value: null,
      type: FORM_TYPES.boolean
    },
    onPress: {
      label: "OnPress handler",
      description: "Function that is called when radio button pressed",
      editable: true,
      required: false,
      value: "{this.onPress}",
      type: FORM_TYPES.function
    }
  },
  layout: {
    width: 343,
    height: 59
  }
};
