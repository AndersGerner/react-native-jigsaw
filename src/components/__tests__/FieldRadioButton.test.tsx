import * as React from "react";
import renderer from "react-test-renderer";
import FieldRadioButton from "../FieldRadioButton";

it("renders field radio button", () => {
  const tree = renderer
    .create(<FieldRadioButton color="blue" selected />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
