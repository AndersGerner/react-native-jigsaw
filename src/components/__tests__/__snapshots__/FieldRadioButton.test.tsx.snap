// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders field radio button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View>
          <View />
        </View>
      </View>
      <Text
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontFamily": "System",
              "fontSize": 16,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 26,
            },
            Object {
              "color": "rgba(90, 69, 255, 1)",
              "marginLeft": 12,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;
